xen_pt: Support UPT for IGD

When passing through an IGD in UPT (universal passthrough) mode rather
than legacy, it does not need to be treated specially.  Fix the
incorrect assumption that any device at position 00:02.0 is an IGD being
passed through in legacy mode.
diff --git a/hw/xen/xen_pt.c b/hw/xen/xen_pt.c
index 006b5b55f249..5d6b8eef346a 100644
--- a/hw/xen/xen_pt.c
+++ b/hw/xen/xen_pt.c
@@ -850,7 +850,8 @@ static void xen_pt_realize(PCIDevice *d, Error **errp)
     if ((s->real_device.domain == XEN_PCI_IGD_DOMAIN) &&
         (s->real_device.bus == XEN_PCI_IGD_BUS) &&
         (s->real_device.dev == XEN_PCI_IGD_DEV) &&
-        (s->real_device.func == XEN_PCI_IGD_FN)) {
+        (s->real_device.func == XEN_PCI_IGD_FN) &&
+        has_igd_gfx_passthru) {
         if (!is_igd_vga_passthrough(&s->real_device)) {
             error_setg(errp, "Need to enable igd-passthru if you're trying"
                     " to passthrough IGD GFX");
